string.go

```diff
@@ -501,6 +501,68 @@ func execAppend(db *DB, args [][]byte) redis.Reply {
 	return reply.MakeIntReply(int64(len(bytes)))
 }
 
+// execSetRange overwrites part of the string stored at key, starting at the specified offset.
+// If the offset is larger than the current length of the string at key, the string is padded with zero-bytes.
+func execSetRange(db *DB, args [][]byte) redis.Reply {
+	key := string(args[0])
+	offset, errNative := strconv.ParseInt(string(args[1]), 10, 64)
+	if errNative != nil {
+		return reply.MakeErrReply(errNative.Error())
+	}
+	value := args[2]
+	bytes, err := db.getAsString(key)
+	if err != nil {
+		return err
+	}
+	bytesLen := int64(len(bytes))
+	if bytesLen < offset {
+		diff := offset - bytesLen
+		diffArray := make([]byte, diff, diff)
+		bytes = append(bytes, diffArray...)
+		bytesLen = int64(len(bytes))
+	}
+	for i := 0; i < len(value); i++ {
+		idx := offset + int64(i)
+		if idx >= bytesLen {
+			bytes = append(bytes, value[i])
+		} else {
+			bytes[idx] = value[i]
+		}
+	}
+	db.PutEntity(key, &DataEntity{
+		Data: bytes,
+	})
+	db.AddAof(makeAofCmd("setRange", args))
+	return reply.MakeBulkReply(bytes)
+}
+
+func execGetRange(db *DB, args [][]byte) redis.Reply {
+	key := string(args[0])
+	startIdx, errNative := strconv.ParseInt(string(args[1]), 10, 64)
+	if errNative != nil {
+		return reply.MakeErrReply(errNative.Error())
+	}
+	endIdx, errNative := strconv.ParseInt(string(args[2]), 10, 64)
+	if errNative != nil {
+		return reply.MakeErrReply(errNative.Error())
+	}
+
+	if startIdx > endIdx {
+		return reply.MakeErrReply("ERR end index should be greater than start index")
+	}
+
+	bytes, err := db.getAsString(key)
+	if err != nil {
+		return err
+	}
+
+	if bytes == nil {
+		return reply.MakeNullBulkReply()
+	}
+
+	return reply.MakeBulkReply(bytes[startIdx:endIdx])
```

Comment:
if  startIdx or endIdx is greater than len(bytes), there will be a panic. start an end can be negative or greater than len(bytes). Please see https://redis.io/commands/getrange and execLIndex

Reply:
This problem remains unresolved. Besides, in redis  start and end both are inclusive while in golang end is exclusice

```
if startIdx < -1*size {
    return &reply.NullBulkReply{}
} else if startIdx < 0 {
    startIdx = size + startIdx
} else if startIdx >= size {
    return &reply.NullBulkReply{}
}
if endIdx < -1*size {
    return &reply.NullBulkReply{}
} else if endIdx < 0 {
    endIdx = size + endIdx + 1
} else if endIdx < size {
    endIdx = endIdx + 1
} else {
    endIdx = size
}
if startIdx > endIdx {
    return reply.MakeNullBulkReply()
}
```